stats/postgres_data_access.go

```diff
@@ -0,0 +1,172 @@
+
+	return da.updateFacts(patient, condition, "increment")
+}
+
+// RemoveConditionStat decrements the relevant county and subdivision condition facts
+// based on the given patient's address.
+func (da *PgStatsDataAccess) RemoveConditionStat(patient *models.Patient, condition *models.Condition) (err error) {
+
+	return da.updateFacts(patient, condition, "decrement")
+}
+
+// identifyCountyAndSubdivisionForPatient returns the countyfp and cousubfp that
+// match the subdivision in the given patient's address.
+func (da *PgStatsDataAccess) identifyCountyAndSubdivisionForPatient(patient *models.Patient) (countyfp, cousubfp string, err error) {
+
+	if len(patient.Address) == 0 || patient.Address[0].City == "" {
+		err = ErrNoAddress
+		return
+	}
+
+	err = da.DB.QueryRow("SELECT countyfp, cousubfp FROM tiger.cousub WHERE name = $1", patient.Address[0].City).Scan(&countyfp, &cousubfp)
+	return
+}
+
+// identifyDiseaseForCondition returns the diseasefp that matches the given
+// condition's SNOMED code, if any.
+func (da *PgStatsDataAccess) identifyDiseaseForCondition(condition *models.Condition) (diseasefp string, err error) {
+	snomedCode, err := getSnomedCode(condition)
+
+	if err != nil || snomedCode == "" {
+		return "", ErrNoSnomedCode
+	}
+	err = da.DB.QueryRow("SELECT diseasefp FROM synth_ma.synth_disease WHERE code_snomed = $1", snomedCode).Scan(&diseasefp)
+	return
+}
+
+// updateStats increments or decrements a row of population counts in the county and subdivision stats tables.
+func (da *PgStatsDataAccess) updateStats(patient *models.Patient, op string) (err error) {
+
+	countyfp, cousubfp, err := da.identifyCountyAndSubdivisionForPatient(patient)
+
+	if err != nil {
+		return
+	}
+
+	if !patientGenderIsValid(patient) {
+		err = ErrInvalidGender
+		return
+	}
+
+	var symbol string
+	switch op {
+	case "increment":
+		symbol = "+"
+	case "decrement":
+		symbol = "-"
+	default:
+		return ErrInvalidUpdateOperation
+	}
+
+	var ctfp, csfp int
+	countyQuery := fmt.Sprintf("UPDATE synth_ma.synth_county_stats SET pop = pop %s 1, pop_%s = pop_%s %s 1, pop_sm = ((pop %s 1) / sq_mi) WHERE ct_fips = $1 RETURNING ct_fips", symbol, patient.Gender, patient.Gender, symbol, symbol)
+	cousubQuery := fmt.Sprintf("UPDATE synth_ma.synth_cousub_stats SET pop = pop %s 1, pop_%s = pop_%s %s 1, pop_sm = ((pop %s 1) / sq_mi) WHERE cs_fips = $1 RETURNING cs_fips", symbol, patient.Gender, patient.Gender, symbol, symbol)
+	err = da.DB.QueryRow(countyQuery, countyfp).Scan(&ctfp)
+	err = da.DB.QueryRow(cousubQuery, cousubfp).Scan(&csfp)
+	return
+}
+
+// updateFacts increments or decrements a row of population counts in the county and subdivision fact tables.
+func (da *PgStatsDataAccess) updateFacts(patient *models.Patient, condition *models.Condition, op string) (err error) {
+
+	countyfp, cousubfp, err := da.identifyCountyAndSubdivisionForPatient(patient)
+	diseasefp, err := da.identifyDiseaseForCondition(condition)
```

Comment:
The first time you assign `err` is getting overwritten by the second time -- so any error that comes up is getting swallowed.  You need to do an error check after identifying the county and subdivision.